qemu/tests/nic_bonding.py

```diff
@@ -77,34 +87,42 @@ def run(test, params, env):
                     time.sleep(random.randint(1, 30))
                     session_serial.cmd(link_set_cmd % (ifname, "up"))
                     time.sleep(random.randint(1, 30))
-
-        except Exception:
+        except aexpect.ShellProcessTerminatedError:
             transfer_thread.join(suppress_exception=True)
             raise
         else:
             transfer_thread.join()
 
+        logging.info('Cleaning temp file on host')
+        os.remove(host_path)
         logging.info("Failover test 2 with file transfer")
         transfer_thread = utils_misc.InterruptedThread(
-            utils_test.run_file_transfer, (test, params, env))
+            vm.copy_files_from, (guest_path, host_path))
         transfer_thread.start()
         try:
             nic_num = len(ifnames)
-            up_index = 0
+            index = 0
             while transfer_thread.isAlive():
-                up_index = up_index % nic_num
-                for num in range(nic_num):
-                    if num == up_index:
-                        session_serial.cmd(link_set_cmd % (ifnames[num], "up"))
-                    else:
-                        session_serial.cmd(link_set_cmd % (ifnames[num], "down"))
-                time.sleep(random.randint(1, 5))
-                up_index += 1
-        except Exception:
+                index = index % nic_num
+                for i in range(nic_num):
+                    session_serial.cmd(link_set_cmd % (ifnames[i], "up"))
+                    for j in range(nic_num):
+                        if i != j:
+                            session_serial.cmd(
+                                link_set_cmd % (ifnames[j], "down"))
+                time.sleep(random.randint(3, 5))
+                index += 1
+        except aexpect.ShellProcessTerminatedError:
             transfer_thread.join(suppress_exception=True)
             raise
         else:
             transfer_thread.join()
+        current_md5 = crypto.hash_file(host_path, algorithm="md5")
+        logging.info("md5 value of data current: %s" % current_md5)
+        if original_md5 != current_md5:
```

Comment:
Add this step of comparing the md5 value to your comment message.

Reply:
Hi @PaulYuuu, I have changed it, please help me review it again, thanks.